### DISCUSSION TITLE
Add optional fn/ln to profiles

### DISCUSSION THREAD

**Discussion by dylsell** [deso-protocol/dips#24] [Medium CIPs]
**Problem**

Trust is hugely important to BitClout's success long term, but many lack trust in the product today. 

**Proposed Solution** 

This is by no means a silver bullet to BitClout's trust issues. But, users should be able to optionally add their FN/LN to their profiles. This will allow creators to build more trust with others on bitclout. 

**Solution Details**

There are a lot of places that reference username. For an MVP I don't think it's necessary to render fn/ln in all of these places. It seems meaningful enough to add fn/ln to profile settings (https://bitclout.com/update-profile) and to the full profile page (https://bitclout.com/u/[username]). From there, other apps and potentially the bitclout.com feeds can decide if this is valuable to implement elsewhere. I talk about this in a bit more length starting at 2m in this video: https://vimeo.com/570408722 

Other apps that leverage an optional fn/ln pattern are:

- GitHub
- Instagram
- Twitter

**Success Criteria**

If this is successful, we'll see a strong % of users filling out their FN/LN and bots will be more easily detectible. 

**Open questions**
Should this be implemented at the core of the user profile and accessible via all user profile APIs? 

**Future iterations**

- Show fn/ln in other places in the bitclout UI outside of the user's profile and profile settings 
- Index fn/ln to be searched on

**Comment by tijno** [comment]
Can you post screenshot of example so i dont need to watch the video ? 

Sorry for dumb question - but what does fn/ln mean ? 

If its links to proven profiles on other sites (login with github to show gh badge) then this makes sense but should probably be considered in light of move to distributed "verification" / associations model described by DH previously in FAQ ([under moving blocks verification to chain](https://docs.bitclout.com/faq/bitclout-faq#roadmap))

**Reply by tijno** [reply]
Ah lol got it

FirstName / LastName?

If so:

i am not in favour of this.

bitclout at its core is a pseudonymous social network and moves away from the trading of forced real name and single accounts (clubhouse / facebook).

in addition as there is no way to verify the info - it is easily faked by bots and therefore not useful in avoiding bots.

**Reply by dylsell** [reply]
Thanks for your thoughts @tijno. Yes, first name last name :) 

Re: 
>forced real name 

nothing is forced nor should be, this is why I included "optional" in the title 

-- 

Re: info verification, that's not supported anywhere nor should it be here. It just gives users who are considering investing or following another data point to conduct research. If I am on the fence about someone, can I google them to do more research? In this current environment, no. Unless somebody adds their fn/ln to their bio which I have seen before organically, which to me is a strong signal that it's important to  some users.

**Reply by HPaulson** [reply]
@tijno Interesting points here -- I actually disagree. I think having this as an entirely optional field would be a cool display feature -- and is even one we tried implementing with Prism using ExtraData on ProfileUpdate transactions back in June. Since it's entirely optional, is this not a positive addition? Some apps could utilize such un unique ways beyond just showing it on your profile. Other social media do this, Instagram in particular, and people use fake names in the field often. Such wouldn't be a source for verification, but just as a nice addition to bio of sorts.

Also, +1 for GH or other badges, but think this is a separate topic

**Reply by tijno** [reply]
Good point @dylsell - it does indeed say optional.

My bad. Sorry. 

I guess some additional aspects to consider:

* First Name / Last Name / Full Name still imply to me people are expected to enter their Real Name

* When someone does not fill it out - will that result in pressure from the community as to why they havent and therefore "they must be a scam"

* People can add their name to their profile right now ? Are people doing this ? I have not seen examples. Does that mean there is no need ? 

* If this change does go ahead can we make it a single string field not named "full name" or "real name". Maybe "know as" or "creator name" or "long name" ?

* There may be a desire to keep adding fields and data to a user profile to help with xyz. I am worried about that and that this change may set a presedent for that. Lots of fields on profiles increases friction for new users, and im not sure adds huge amount of value over the basic profile.

For me simplicity will remain key as Bitclout is quit complez already. 

@HPaulson - i like the idea of ExtraData on User profiles enabling structured data that nodes and apps can build on.

**Reply by dylsell** [reply]
>When someone does not fill it out - will that result in pressure from the community as to why they havent and therefore "they must be a scam"

- I'd hope not, there are plenty of other signals that an account is legit. See "fuckjerry" on IG as the first that comes to mind. Quality content can trump anything else. 

>People can add their name to their profile right now ? Are people doing this ? I have not seen examples. Does that mean there is no need ?

- I've seen at least one of these in the wild, will re-surface it if I find it again. Bio space is already v limited though. 

> If this change does go ahead can we make it a single string field not named "full name" or "real name". Maybe "know as" or "creator name" or "long name" ?

- This solves the same problem IMO so 👍 regardless of how this is implemented. 

>There may be a desire to keep adding fields and data to a user profile to help with xyz. I am worried about that and that this change may set a presedent for that. Lots of fields on profiles increases friction for new users, and im not sure adds huge amount of value over the basic profile.

- This'll be true for anything we add anywhere. That's where a strong product vision from the core devs will be required to sort out what's needed and what's fluff.

**Reply by HPaulson** [reply]
> People can add their name to their profile right now ? Are people doing this ? I have not seen examples. Does that mean there is no need ?

lol, https://bitclout.com/u/HPaulson

**Reply by tijno** [reply]
Ah you are exceptional hunter. Should have known it was in there. 

Lets see how the discussion evolves.

**Reply by AikonMishima** [reply]
The idea makes sense as it’s an optional field.

There’s different perceptions at play, however, when it comes to the pseudonymous nature of BitClout and entering a real name.

The more positive, would be the introduction of the account’s verification of authenticity (note: perception, not technical authentication).
Another positive is the ability to speak to people’s first name, instead of a nickname.
And in final, the usage of this data for making communication more « personal ».

From the other end of the spectrum, it could be used to reinforce impersonation by scammers / bots.

In conclusion: I’m all for it, just that it needs to be well motivated for the main node.
As in: « Filling in your real name is NOT a necessity, it’s optional. »

**Reply by dylsell** [reply]
Here is a good example of how Reddit enabled an optional display name alongside a username 

<img width="728" alt="Screen Shot 2021-07-12 at 9 30 13 PM" src="https://user-images.githubusercontent.com/33188041/125376163-72591500-e358-11eb-8c07-4d343779af38.png">

@diamondhands0 @maebeam now that this discussion has evolved a bit, would love your thoughts

**Reply by tijno** [reply]
"display name (optional)" is a good way. Id be happy with that. 

Still concerned about this potentially making impersonation easier.

Username has to be unique so scammers have to resort to using lookalike usernames for impersonation.

**Comment by oO** [comment]
separate fields for First Name + Last Names is a very narrow Western Culture format that doesn't fit a lot of other naming formats ( chinese names have 3 parts, for example )

So while I think having a standard field for a "Display Name" ( real or invented ) with less restrictions on characters in addition to the @username ( which has to be globally unique ) is a good idea, I'm definitely against the First+Last separation.

Also look at Supernovas.app as an example of a node that already supports that concept. They just don't store the information on-chain, but in their own extension to the database.

![Screen Shot 2022-02-22 at 10 52 30 AM](https://user-images.githubusercontent.com/181864/155199477-3d6da95e-fb50-4919-8e69-d2e83ba2ca6d.png)
